techmatters/flex-plugins
# CHI-1526: Optimised flex log proxying to only proxy errors.

Primary reviewer: @robert-bo-davis 

## Description

The rollbar proxing change is to address performance issues observed since the migration to Flex 2, where large amounts of browser console logging lead to high resource usage & slow updates due to work being done when proxying console.log calls to the Rollbar SDK

* Updated Flex console.log proxying to only proxy 'console.error' to Rollbar, rather than all console mewthods, since the Rollbar client is configured to filter out all log events below 'error' level befopre uploading anyway
* Updated application logic directly driven from NODE_ENV to work off separate environment variables
* Updated webpack to use dotenv-webpack to combine .env files and CLI env vars into browser's 'process.env' 
* Added .env files with defaults based on NODE_ENV
* Converted setUpMonitoring to TS


### Checklist
- N/A New tests added
- N/A Feature flags added
- N/A Strings are localized

### Related Issues
CHI-1526

### Verification steps

* Add a 'console.error' log line to your local version
* Set 'ENABLE_MONITORING=true' for your command line environment
* Run the plugin locally
* Check the error is reported in Rollbar
* Run the plugin again without the 'ENABLE_MONITORING' environment variable dset
* Check the error is NOT reported
